emva1288/unittests/test_results.py

```diff
@@ -300,3 +302,68 @@ def test_nans(self):
         self.assertIs(r.DSNU1288_DN(), np.nan)
 
         del r
+
+
+class TestRoutines(unittest.TestCase):
+
+    # instantiate a bayer filter and a wrong filter
+    bayer_filter = get_bayer_filter(0, 1, 1, 0, 1000, 1000)
+    wrong_filter = np.tile([[1, 1, 1, 0], [1, 0, 0, 1]], (500, 250))
+
+    # intantiate masked images based in the masks
+    img = np.abs(np.random.random([1000, 1000]))
+    wrong = np.ma.array(img, mask=wrong_filter)
+    masked = np.ma.array(img, mask=bayer_filter)
+
+    # save the filtered images, masked and with no mask
+    res = high_pass_filter(img, 5)
+    img_filt = res['img']
+    im_num = res['multiplicator']
+    res = high_pass_filter(masked, 5)
+    ma_filt = res['img']
+    ma_num = res['multiplicator']
+
+    # introduce a low frequency signal in the images
+    sig = np.sin(np.linspace(0, np.pi, 1000))
+    sig_img = [line + val for line, val in zip(img, sig)]
+    sig_masked = np.ma.array(sig_img, mask=bayer_filter)
+
+    # filtering the images with a low frequency signal
+    res = high_pass_filter(sig_img, 5)
+    si_filt = res['img']
+    si_num = res['multiplicator']
+    res = high_pass_filter(sig_masked, 5)
+    sm_filt = res['img']
+    sm_num = res['multiplicator']
+
+    def test_highpass_filter(self):
+        # Assert raises in case of bad kernel size or bad mask
+        with self.assertRaises(ValueError):
+            high_pass_filter(self.wrong, 5)
+        with self.assertRaises(ValueError):
+            high_pass_filter(self.img, 6)
+
+        # assert that the means of the filtered images are almost 0
+        self.assertAlmostEqual(0, np.mean(self.ma_filt/self.ma_num), 3)
+        self.assertAlmostEqual(0, np.mean(self.img_filt/self.im_num), 3)
+
+        # Assert that the std stays about the same after filtering an image
```

Comment:
shouldn't be the std from the original image used everywhere?

Reply:
Yes, and the test does just that. For the sig_img_filt and sig_imgc_filt, we use respectively img_filt and imgc_filt, which have already been compared the img and imgc. Having already compared img to img_filt with a 10e-1 criteria, comparing img_filt to sig_img_filt with a criteria of 10e-4 compares sig_img_filt to img with a criteria of 10e-1, and also asserts that filtering img and sig_img gives the same result, which is a desired behavior. Overall, it makes the test more accurate. I could change it if its not clear.